gems/sorbet-runtime/lib/types/types/typed_module.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+# typed: true
+
+module T::Types
+  class TypedModule < T::Types::Base
+    def initialize(type)
+      @inner_type = type
+    end
+
+    def type
+      @type ||= T::Utils.coerce(@inner_type)
+    end
+
+    def build_type
+      type
+      nil
+    end
+
+    # overrides Base
+    def name
+      "T::Module[#{type.name}]"
+    end
+
+    def underlying_class
+      Module
+    end
+
+    # overrides Base
+    def valid?(obj)
+      Module.===(obj)
+    end
+
+    # overrides Base
+    private def subtype_of_single?(type)
+      case type
+      when TypedModule
+        # treat like generics are erased
+        true
+      when Simple
+        Module <= type.raw_type
+      else
+        false
+      end
+    end
+
+    module Private
+      module Pool
+        CACHE_FROZEN_OBJECTS =
+          begin
+            ObjectSpace::WeakMap.new[1] = 1
+            true # Ruby 2.7 and newer
+          rescue ArgumentError
+            false # Ruby 2.6 and older
```

Comment:
Should we include this, now that we officially don't support ruby 2.x?

Reply:
Pretty sure I cargo culted this from the type_class.rb file. We can consider removing them in another PR I guess, though I don't mind keeping it.